crypto/x509/x_all.c

```diff
@@ -138,12 +175,60 @@ X509 *X509_load_http(const char *url, BIO *bio, BIO *rbio, int timeout)
                                    ASN1_ITEM_rptr(X509));
 }
 
+/*
+ * Drop empty SKID or AKID X509_REQ extensions that might result from:
+ * "...KeyIdentifier = none"
+ */
+
+static void drop_empty_req_keyid_exts(X509_REQ *x)
+{
+    STACK_OF(X509_EXTENSION) *exts;
+    int i, n;
+
+    if ((exts = ossl_x509_req_get1_extensions_by_nid(x, NID_ext_req)) == NULL
+        || (n = sk_X509_EXTENSION_num(exts)) < 1) {
+        sk_X509_EXTENSION_pop_free(exts, X509_EXTENSION_free);
+        return;
+    }
+
+    for (i = 0; i < n; ++i) {
+        X509_EXTENSION *ext = sk_X509_EXTENSION_value(exts, i);
+        ASN1_STRING *der = X509_EXTENSION_get_data(ext);
+
+        /*
+         * Empty OCTET STRINGs and empty SEQUENCEs encode to just two bytes of
+         * tag (0x04 or 0x30) and length (0x00).
+         */
+        if (der->length == 2 && (der->data[0] == 0x04 || der->data[0] == 0x30)) {
```

Comment:
```suggestion
        if (ASN1_STRING_length(der) == 0) { /* indicating "none" */
```

Reply:
It is however simply wrong. This ASN.1 string is a DER encoding of the **extension**, it is two bytes long when the extension payload is an empty octet string or an empty sequence. The code works as written. I am marking this comment as resolved, even if you happen to disagree. If you still care, take some time to check the behaviour by stepping through the function in a debugger.